lib/manager/containers/ActiveProjectViewer.js

```diff
@@ -62,29 +62,29 @@ const mapDispatchToProps = (dispatch, ownProps) => {
-    updateProjectSettings: (project, newSettings) => dispatch(updateProject(project, newSettings, true)),
-    thirdPartySync: (type) => dispatch(thirdPartySync(projectId, type)),
-    updateAllFeeds: (project) => dispatch(fetchFeedsForProject(project)),
-    saveFeedSource: (name) => dispatch(saveFeedSource({ projectId, name })),
-    updateFeedSourceProperty: (feedSource, propName, newValue) => dispatch(updateFeedSource(feedSource, { [propName]: newValue })),
-    deploymentsRequested: () => dispatch(fetchProjectDeployments(projectId)),
-    deployPublic: (project) => dispatch(deployPublic(project)),
     createDeploymentFromFeedSource: (feedSource) => dispatch(createDeploymentFromFeedSource(feedSource)),
-    onNewDeploymentClick: () => dispatch(createDeployment(projectId)),
-    newDeploymentNamed: (name) => dispatch(saveDeployment({ projectId, name })),
-    updateDeployment: (deployment, changes) => dispatch(updateDeployment(deployment, changes)),
-    searchTextChanged: (text) => dispatch(setVisibilitySearchText(text)),
-    visibilityFilterChanged: (filter) => dispatch(setVisibilityFilter(filter)),
-    uploadFeed: (feedSource, file) => dispatch(uploadFeed(feedSource, file)),
-    fetchFeed: (feedSource) => dispatch(runFetchFeed(feedSource)),
     deleteFeedSource: (feedSource) => dispatch(deleteFeedSource(feedSource)),
     deleteProject: (project) => {
       return dispatch(deleteProject(project))
         .then(() => browserHistory.push('/home'))
     },
     deleteDeploymentConfirmed: (deployment) => dispatch(deleteDeployment(deployment)),
-    downloadMergedFeed: (project) => dispatch(downloadFeedForProject(project))
+    deploymentsRequested: () => dispatch(fetchProjectDeployments(projectId)),
+    deployPublic: (project) => dispatch(deployPublic(project)),
+    downloadMergedFeed: (project) => dispatch(downloadFeedForProject(project)),
+    fetchFeed: (feedSource) => dispatch(runFetchFeed(feedSource)),
+    newDeploymentNamed: (name) => dispatch(saveDeployment({ projectId, name })),
+    onNewDeploymentClick: () => dispatch(createDeployment(projectId)),
+    onNewFeedSourceClick: () => dispatch(createFeedSource(projectId)),
+    saveFeedSource: (name) => dispatch(saveFeedSource({ projectId, name })),
+    searchTextChanged: (text) => dispatch(setVisibilitySearchText(text)),
+    thirdPartySync: (type) => dispatch(thirdPartySync(projectId, type)),
+    updateFeedSourceProperty: (feedSource, propName, newValue) => dispatch(updateFeedSource(feedSource, { [propName]: newValue })),
+    updateAllFeeds: (project) => dispatch(fetchFeedsForProject(project)),
+    updateDeployment: (deployment, changes) => dispatch(updateDeployment(deployment, changes)),
+    updateProjectSettings: (project, newSettings) => dispatch(updateProject(project, newSettings, true)),
+    uploadFeed: (feedSource, file) => dispatch(uploadFeed(feedSource, file)),
+    visibilityFilterChanged: (filter) => dispatch(setVisibilityFilter(filter))
```

Comment:
At some point, we should refactor all of the mapDispatchToProps actions.  This is a lot of unneeded boilerplate.

Reply:
Yes, pretty much the only thing keeping that from happening is the `onComponentMount`, which has some custom logic that depends on dispatching actions.  But maybe this needs to be defined as its own action (e.g., `onActiveProjectViewerMount`).  Creating those can be a new PR.